File: matlab/phasor_circuit_example4.m
clear all

v1_magnitude = 240;
v1_phase = 53.13*pi/180;
v2_magnitude = 96;
v2_phase = 0*pi/180;
V1 = v1_magnitude*cos(v1_phase) + j*v1_magnitude*sin(v1_phase);
V2 = v2_magnitude*cos(v2_phase) + j*v2_magnitude*sin(v2_phase);
Z1 = j*4000*15e-6;
Z2 = 30;
Z3 = -j/4000/25e-6;
Z4 = 20;

Zeq = parallel(parallel(parallel(Z1, Z2), Z3), Z4);
Ieq = V1/Z1 - V2/Z4;
V0 = Ieq*Zeq;
v0_magnitude = abs(V0);
v0_phase = angle(V0) *180/pi;


function Zeq = parallel(Z1, Z2)
    Zeq = Z1*Z2/(Z1+Z2);
end

File: python/phasor_circuit_example4.py
import cmath
import numpy as np

def parallel(Z1, Z2):
    Zeq = Z1 * Z2 / (Z1 + Z2)
    return Zeq

v1_magnitude = 10
v1_phase = 0*np.pi/180
V1 = cmath.rect(v1_magnitude, v1_phase)
v2_magnitude = 100
v2_phase = -90*np.pi/180
V2 = cmath.rect(v2_magnitude, v2_phase)
Z1 = 50
Z2 = -1j/50000/9e-6
Z3 = 1j*50000*100e-6
Z4 = 20
Ieq = I1 - V2/Z4
Zeq = parallel(Z1, parallel(Z2, parallel(Z3, Z4)))
V0 = Ieq * Zeq
V0_polar = cmath.polar(V0)
v0_magnitude = V0_polar[0]
v0_phase = V0_polar[1] * 180/np.pi
foo = 1